wrangler-core/src/main/java/io/cdap/functions/JsonFunctions.java

```
@@ -0,0 +1,169 @@
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ *  WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ *  License for the specific language governing permissions and limitations under
+ *  the License.
+ */
+
+package io.cdap.functions;
+
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import com.google.gson.JsonArray;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonNull;
+import com.google.gson.JsonParser;
+import com.google.gson.JsonSyntaxException;
+import com.jayway.jsonpath.Configuration;
+import com.jayway.jsonpath.DocumentContext;
+import com.jayway.jsonpath.JsonPath;
+import com.jayway.jsonpath.Option;
+import com.jayway.jsonpath.spi.json.GsonJsonProvider;
+import com.jayway.jsonpath.spi.mapper.GsonMappingProvider;
+
+/**
+ * Collection of useful expression functions made available in the context
+ * of an expression.
+ *
+ * set-column column <expression>
+ */
+public final class JsonFunctions {
+  public static final Configuration GSON_CONFIGURATION = Configuration
+    .builder()
+    .mappingProvider(new GsonMappingProvider())
+    .jsonProvider(new GsonJsonProvider())
+    .options(Option.SUPPRESS_EXCEPTIONS)
+    .build();
+
+  private static final JsonParser PARSER = new JsonParser();
+  private static final Gson GSON = new GsonBuilder().serializeNulls().create();
+
+  /**
+   * Don't let anyone instantiate this class.
+   */
+  private JsonFunctions() {}
+
+  /**
+   * Parses a column or string to JSON. This is equivalent to <code>JSON.parse()</code>
+   *
+   * @param json string representation of json.
+   * @return parsed json else throws an exception.
+   */
+  public static JsonElement Parse(String json) {
+    try {
+      JsonElement element = PARSER.parse(json);
+      return element;
+    } catch (JsonSyntaxException e) {
+      return JsonNull.INSTANCE;
```

This silently returns null json element. should this log any error message?

---

Not really, because this happens in parsing of data, it will fill the logs.